src/main/java/courseManager/InstanceList.java

```diff
@@ -0,0 +1,143 @@
+package courseManager;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class InstanceList {
+    private ArrayList<AssessmentInstance> listOfAssessmentInstances;
+
+    private String pluralTitle;
+    private String singularTitle;
+    private int totalNumberOfInstances;
+    private int numberOfCommittedInstances;
+    private int numberOfSubmittedInstances;
+
+    /**
+     * Create a new InstanceList object with a list of AssessmentInstances
+     * @param pluralTitle the plural title of the assessment
+     *              e.g. "Quizzes", "Midterm", "Homework Assignments"
+     * @param totalNumberOfInstances the total number of instances of the assessment
+     */
+
+    public InstanceList(String pluralTitle, int totalNumberOfInstances) {
+        this.listOfAssessmentInstances = new ArrayList<>();
+        this.singularTitle = toSingular(pluralTitle);
+        for (int i = 0, j = 1; i < totalNumberOfInstances; i++, j++){
+            this.addInstance(this.singularTitle + " #" + j);
```

Comment:
Consider adding another constructor that allows a user to initialize an assessment with 0 instances. This would be preferable for people who like keeping track of things "as they go" and would require very minimal modification (I think addInstance just needs a check to see if we've already reached the total allowed.)

Reply:
In parallel to this, the totalNumberOfInstances should be removed from addInstance, since if a user only has 3 out of 5 instances we would still want to say the total is 5. A "getCurrentNumberOfInstances" function can also be added.

Reply:
I don't 100% understand what you mean here, since the totalNumberOfInstances is prescribed by the Assessment's associated weightscheme. As I imagined it so far, InstanceList is automatically generated through the Assessment's constructor.